site/content/in-dev/unreleased/configuring-polaris-for-production.md

```diff
@@ -23,114 +23,179 @@ type: docs
+your own `persistence.xml` file. This file contains details of the database used for metastore
+management and the connection settings. For more information, refer to the [metastore
+documentation]({{% ref "metastores" %}}).
 
-```bash
-bin/polaris-service bootstrap config/polaris-server.yml
+Then, configure Polaris to use your metastore by setting the following properties:
+
+```properties
+polaris.persistence.type=eclipse-link
+polaris.persistence.eclipselink.configuration-file=/path/to/persistence.xml
+polaris.persistence.eclipselink.persistence-unit=polaris
 ```
 
-Afterward, Polaris can be launched normally:
+Where:
 
-```bash
-java -jar /path/to/jar/polaris-service-all.jar server polaris-server.yml
+- `polaris.persistence.type` indicates that we are using the EclipseLink metastore.
+- `polaris.persistence.eclipselink.configuration-file` is the path to the `persistence.xml` file.
+- `polaris.persistence.eclipselink.persistence-unit` is the name of the persistence unit to use (in 
+  case the configuration file has many persistence units).
+
+Typically, in Kubernetes, you would define the `persistence.xml` file as a `ConfigMap` and set the
+`polaris.persistence.eclipselink.configuration-file` property to the path of the mounted file in
+the container.
+
+Be sure to secure your metastore backend since it will be storing credentials and catalog metadata.
+
+### Bootstrapping
+
+Before using Polaris, you must **bootstrap** the metastore manager. This is a manual operation that
+must be performed **only once** in order to prepare the metastore manager to integrate with Polaris.
+
+By default, when bootstrapping a new realm, Polaris will create randomised `CLIENT_ID` and
+`CLIENT_SECRET` for the `root` principal and store their hashes in the metastore backend.
+
+Depending on your database, this may not be convenient as the generated credentials are not stored
+in clear text in the database.
+
+In order to provide your own credentials for `root` principal (so you can request tokens via
+`api/catalog/v1/oauth/tokens`), there are two approaches:
+
+1. Use the Polaris Admin Tool to bootstrap the realm and set its `root`
+   principal credentials.
+2. Set the `root` principal credentials when deploying Polaris for the first time.
+
+The first approach is recommended as it is more flexible (you can bootstrap new realms at any time
+without incurring in a Polaris downtime). The second approach is useful for testing and development,
+but can be used in production as well.
+
+In order to bootstrap root credentials for a realm named `my-realm` when deploying Polaris, set the
+following environment variables:
+
+```
+export POLARIS_BOOTSTRAP_CREDENTIALS=my-realm,my-client-id,my-client-secret
 ```
 
+If the realm hasn't been bootstrapped yet, Polaris will create the realm and the `root` principal
+with the provided credentials upon first usage of that realm. If the realm already exists, Polaris
+will not attempt to update the `root` principal credentials.
```

Comment:
Does it mean the bootstrap will fail if the realm exists already? If that's so, we may not mention it or just say you cannot bootstrap an existing realm.

Reply:
I just checked and indeed it currently fails, and quite inelegantly. I will raise a PR shortly to improve the user experience.

In any case, this paragraph will disappear from this file since we agreed above to only mention option 1.